cmake/CMakeLists.txt

```diff
@@ -76,19 +78,71 @@ set(PUBLIC_HEADERS
   ${ROOT_DIR}/include/winsparkle.h
   ${ROOT_DIR}/include/winsparkle-version.h)
 
-set(RESOURCES winsparkle translations)
-
-# resource compilation for MinGW
-foreach(RESOURCE ${RESOURCES})
-  add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${RESOURCE}.o
-                     COMMAND ${CMAKE_RC_COMPILER} -I${CMAKE_CURRENT_SOURCE_DIR} -i${CMAKE_CURRENT_SOURCE_DIR}/${RESOURCE}.rc
-                             -o ${CMAKE_CURRENT_BINARY_DIR}/${RESOURCE}.o)
+set(LANGUAGES
+  ar
+  be
+  bg
+  bs
+  ca_ES
+  co
+  cs
+  da
+  de
+  el
+  es
+  eu
+  fr
+  fy_NL
+  ga
+  he
+  hr
+  hu
+  hy
+  id
+  it
+  ja
+  kk
+  ko
+  nb
+  nl
+  pl
+  pt_BR
+  pt_PT
+  ru
+  sk
+  sr
+  sv
+  tr
+  uk
+  zh_CN
+  zh_TW
+)
+
+foreach(LANG ${LANGUAGES})
+  add_custom_command(OUTPUT ${LANG}.mo
+                     COMMAND ${GETTEXT_MSGFMT_EXECUTABLE} -c -o ${LANG}.mo ${ROOT_DIR}/translations/${LANG}.po
+                     DEPENDS ${ROOT_DIR}/translations/${LANG}.po
+  )
+
+  list(APPEND MO_FILES ${LANG}.mo)
 endforeach()
 
+add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/common_res.o
+                   COMMAND ${CMAKE_RC_COMPILER} -O coff -I${ROOT_DIR}/include -i ${CMAKE_CURRENT_SOURCE_DIR}/common.rc
+                           -o ${CMAKE_CURRENT_BINARY_DIR}/common_res.o
+                   DEPENDS ${MO_FILES}
+)
+
+list(APPEND SOURCES ${CMAKE_CURRENT_BINARY_DIR}/common_res.o)
+
 add_library(${PROJECT_NAME} SHARED ${SOURCES} $<TARGET_OBJECTS:wxWidgets> $<TARGET_OBJECTS:expat>)
 
 target_link_libraries(${PROJECT_NAME} wininet version rpcrt4 comctl32)
 
+#set_target_properties(${PROJECT_NAME} PROPERTIES
```

Comment:
Please don't include commented out code — either it is needed, in which case it shouldn't be commented out, or it isn't, in which case it should be there at all.

Reply:
Fixed